src/workloads/contrib/historystore/eMRCf_runner.sh

```diff
@@ -0,0 +1,230 @@
+    ./scripts/genny run --workload-file $job --mongo-uri 'mongodb://localhost:27017'
+
+    mv build/genny-metrics $metrics_dir/$name
+    mv build/genny-metrics.csv $metrics_dir/$name/$name.csv
+    ls -l $db_dir/node1/db > $metrics_dir/$name/$name.ls.out
+    mongo --eval 'db.runCommand({collStats: "Collection0"})' > $metrics_dir/$name/$name.collstats
+}
+
+emrcf_opt="--enableMajorityReadConcern true"
+grow_count=10
+kill_secondary=yes
+plot=no
+# Parse command line arguments
+while :; do
+    case "$1" in
+    -b)
+        kill_secondary=no
+        shift ;;
+    -f)
+        emrcf_opt="--enableMajorityReadConcern false"
+        shift ;;
+    -g)
+        grow_count=$2
+        shift ; shift ;;
+    -m)
+        metrics_dir=$2
+        shift ; shift ;;
+    -p)
+        plot=yes
+        gnuplot_cmd=`which gnuplot`
+        if [ ! -x "$gnuplot_cmd" ]
+        then
+            echo "-p option requires gnuplot"
+            exit 1
+        fi
+        shift ;;
+    -*)
+        usage ;;
+     *)
+        break ;;
+    esac
+done
+
+        
+populate=~/proj/eMRCfPopulate.yml
+grow=~/proj/eMRCfGrow.yml
+benchmark=~/proj/eMRCfBench.yml
+
+# Now we run the workload.
+
+echo "Setting up PSA replica set ..."
+mongo_opts="--setParameter enableFlowControl=false $emrcf_opt"
+start_psa
+
+# Genny prep
+mkdir -p $metrics_dir &> /dev/null
+cd $metrics_dir
+metrics_dir=$PWD
+
+cd $genny_dir
+
+run_genny $populate Populate
+
+if [ "$kill_secondary" = "yes" ]
+then
+    echo "Shutting down secondary node ..."
+    mongod --dbpath $db_dir/node2/db --shutdown &> /dev/null
+fi
+
+# Let any delayed processing quiesce then run the benchmark
+sleep 120
+run_genny $benchmark Bench_initial
+
+# Grow the history store
+for X in $(seq 1 $grow_count)
+do
+    run_genny $grow Grow_$X
+done
+
+# Let any delayed processing from the grow phase quiesce then re-run the benchmark
+sleep 120
+run_genny $benchmark Bench_final
+
+# Copy additional data from run.
+cp -R $db_dir/node1/db/diagnostic.data $metrics_dir
+cp -R $db_dir/node1/db/mongod.log $metrics_dir
+
+if [ "$plot" = "no" ]
+then
+    # Don't need to plot results.  We're done.
+    exit 0
+fi
+
+cd $metrics_dir
+
+# For each benchmark phase, we pull the relevant operations out of the CSV files and sort them
+# to generate a CDF.  
+#
+# Note that the csv files created by Genny capture per request latency in nanoseconds.
+# We divide by 1000 and plot them in microseconds.
+
+echo "Preparing plots..."
+tmpfile=$(mktemp)
+for RUN in initial final
+do
+    mkdir $RUN &> /dev/null
+    for PHASE in InsertPhase UpdateHeavyPhase ReadMostlyPhase DeletePhase
+    do
+        echo "    Processing $RUN/$PHASE..."
+        grep "^[0-9].*$PHASE.*One" Bench_$RUN/Bench_$RUN.csv > $tmpfile
+        count=$(wc -l $tmpfile | awk '{print $1}')
+        count=$(($count / 100))
+        cat $tmpfile | sed -e 's/,/ /g' | awk '{print $5}' | sort -n | \
+            awk "BEGIN {n = 1} {print \$1/1000, n/$count; n++}" > $RUN/$PHASE
```

Comment:
I love me a good `awk` script. Unfortunately this analysis logic will be broken soon when we kill the CSV output format. Can you put a comment here or in the readme indicating that the plotting may be broken?

Reply:
Good point.  I'll add comments accordingly.  I assume this means that the `Format: csv-fctdc` metrics option will be going away as well?

Reply:
Correct - there will no longer be an option to specify the output format, it will always be ftdc.